osf/models/files.py

```diff
@@ -445,7 +445,7 @@ def update(self, revision, data, user=None):
                 break
         else:
             # Insert into history if there is no matching etag
-            if data.get('modified'):
+            if data.get('modified') and None not in [item['modified'] for item in self.history]:
```

Comment:
Note from Fitz:

> Okay, this is a weird one. It's possible that we shouldn't be doing the insort at all. It was added 
> early on as part of supporting provenance tracking, but there's no specific message saying 
> why we do it this way. I can't think of a situation that touch() would call update() with anything 
> other than the latest revision data. It would make sense to continue to check the etags, and if 
> not found append the revision to the history.
> 
> One major concern: since this is the update() method for the File class, it might be used in 
> other places in the codebase where the revision isn't the most recent revision (migrations?). 
> Maybe add a debug statement and run the test suite to see who is using this?